geode-core/src/main/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceService.java

```diff
@@ -373,22 +384,28 @@ public boolean removeJars(String[] jarNames, String[] groups) {
           break;
         }
 
-        for (String jarRemoved : jarNames) {
-          File jar = getPathToJarOnThisLocator(group, jarRemoved).toFile();
+        logger.debug("Configuration before removing deployment: " + configuration);
+        Configuration configurationCopy = new Configuration(configuration);
+
+        for (Entry<String, String> deploymentToRemove : deploymentsToRemove.entrySet()) {
+          File jar = getPathToJarOnThisLocator(group, deploymentToRemove.getValue()).toFile();
           if (jar.exists()) {
             try {
               FileUtils.forceDelete(jar);
+              logger.debug("Successfully deleted: " + jar.getName());
+              configurationCopy
+                  .removeDeployments(Collections.singleton(deploymentToRemove.getKey()));
```

Comment:
Just looking at the diff here it appears that removal was happening only once after the loop but that now it happens (potentially) each time through the loop. Not a problem but I just wanted to confirm that's what you wanted.

Reply:
Yes, this was intentional.